### PR TITLE
Add transaction mode for retryTransaction

### PR DESCRIPTION
**Pull request by ilyakharev**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

## Pull request type

Please check the type of change your PR introduces:

- [x] Feature

### CODE REVIEW

File: src/Session.php [hunk 1]
```diff
@@ -218,15 +221,45 @@ public function transaction(Closure $closure)
...
+                break;
+
+            case 'snapshot':
+            case 'snapshot_read_only':
+                $tx_settings['snapshot_read_only'] = new SnapshotModeSettings;
+                break;
+
+            case 'serializable':
+            case 'serializable_read_write':
+            default:
```
Comment (rekby) [inline src/Session.php:257]: Need explicit check null value and raise exception for unknown values - for detect typos.

File: src/Table.php [hunk 2]
```diff
@@ -505,9 +505,15 @@ public function retryTransaction(Closure $userFunc, bool $idempotent = null, Ret
         if (!isset($options['callback_on_error'])) {
             $options['callback_on_error'] = function (\Exception $exception) {};
         }
-        return $this->retrySession(function (Session $session) use ($options, $userFunc) {
+
+        $txMode = '';
```
Comment (rekby) [inline src/Table.php:509]: for example: '' - is not valid transaction mode.

File: src/Session.php [hunk 3]
```diff
@@ -218,15 +221,45 @@ public function transaction(Closure $closure)
      * @return mixed
      * @throws Exception
      */
-    public function beginTransaction()
+    public function beginTransaction(string $mode = null)
```
Comment (rekby) [inline src/Session.php:224]: need tests for the transaction modes.

File: src/Session.php [hunk 4]
```diff
@@ -645,4 +643,52 @@ protected function streamRequest($method, array $data = [])
     {
         return $this->doStreamRequest('Table', $method, $data);
     }
+
+    /**
+     * @param string $mode
+     * @return array
+     * @throws Exception
+     */
+    protected static function parseTxMode(string $mode = null): array
```
Comment (rekby) [inline src/Session.php:652]: no need default value for $mode param

File: tests/CheckTxSettingsTest.php [hunk 5]
```diff
@@ -3,70 +3,46 @@
...
-        $this->session = $this->table->session();
-    }
-
-    public function testSerializableTxConfig(){
-        $this->checkTx('serializable', 'serializable_read_write');
+        foreach ($tests as $i => $test){
+            self::assertEquals($test["result"], Session::parseTxMode($test["mode"]));
+        }
+        self::expectException('Exception');
+        Session::parseTxMode(null);
```
Comment (rekby) [inline tests/CheckTxSettingsTest.php:40]: what about do request with the tx modes?

**Comment by ilyakharev** [comment]
Moved #130

**Pull request closed without merging**